application/app/src/main/java/com/example/buber/DB/DBManager.java

```diff
@@ -208,10 +205,19 @@ public void updateDriver(String docID, Driver updatedDriver, EventCompletionList
                 });
     }
 
-    public void updateTrip(String docID, Trip updatedTrip, EventCompletionListener listener) {
+    public void updateTrip(String docID, Trip updatedTrip, EventCompletionListener listener, boolean listenForUpdates) {
         collectionTrip.document(docID)
                 .set(updatedTrip, SetOptions.merge())
-                .addOnSuccessListener(documentSnapshot -> {
+                .addOnSuccessListener(aVoid -> {
+                    if (listenForUpdates) {
+                        collectionTrip.document(updatedTrip.getRiderID()).addSnapshotListener((documentSnapshot1, e) -> {
```

Comment:
Should be:
```Listener tripListener = collectionTrip....```
then
```App.getModel.setListener(tripListener)```
so that we have some way to unsubscribe the listeners on ```onDestroy()``` of activities.